### PR TITLE
Add linux battery from sysclass

### PR DESCRIPTION
**Pull request by KeyWeeUsr**
Should give us a very stable point for this api on gnu/linux. Haven't checked the exact kernel version, I'll update it in the future.

### CODE REVIEW

File: plyer/platforms/linux/battery.py [hunk 1]
```diff
@@ -2,15 +2,50 @@
...
     '''
-    Implementation of Linux battery API.
+    Implementation of Linux battery API via accessing the sysclass power_supply
+    path from the kernel.
+    '''
+
+    def _get_state(self):
+        status = {"isCharging": None, "percentage": None}
+
+        kernel_bat_path = join('/sys', 'class', 'power_supply', 'BAT0')
```
Comment (tshirtman) [inline plyer/platforms/linux/battery.py]: Some laptops, like mine, have multiple batteries, in my laptop i have BAT0 and BAT1 in this directory, if possible, it would be nice to provide either a list of results, or a merge of the result (total charge in all batteries over total potential). It's a corner case though, so i'm fine with merging as-is.

Comment (KeyWeeUsr) [reply plyer/platforms/linux/battery.py]: If that's the case, please add the correct report from upower either to the test or as a comment, so that I can rework it for that too. I'll merge it as is because the support hasn't been added before anyway, but another PR will have thia feature if there is a upower report available. :)

Comment (tshirtman) [reply plyer/platforms/linux/battery.py]:
```
gabriel@gryphon:~> env LANG=C upower --show-info /org/freedesktop/UPower/devices/battery_BAT1
  native-path:          BAT1
  vendor:               SANYO
  model:                01AV427
  serial:               3199
  power supply:         yes
  updated:              Wed Aug  8 12:21:27 2018 (78 seconds ago)
  has history:          yes
  has statistics:       yes
  battery
    present:             yes
    rechargeable:        yes
    state:               discharging
    warning-level:       none
    energy:              69.11 Wh
    energy-empty:        0 Wh
    energy-full:         69.2 Wh
    energy-full-design:  71.28 Wh
    energy-rate:         9.167 W
    voltage:             12.372 V
    time to empty:       7.5 hours
    percentage:          99%
    capacity:            97.0819%
    technology:          lithium-ion
    icon-name:          'battery-full-symbolic'
  History (charge):
    1533723687	99.000	discharging
  History (rate):
    1533723687	9.167	discharging
```
```
gabriel@gryphon:~> env LANG=C upower --show-info /org/freedesktop/UPower/devices/battery_BAT0
  native-path:          BAT0
  vendor:               SMP
  model:                01AV421
  serial:               1162
  power supply:         yes
  updated:              Wed Aug  8 12:21:27 2018 (81 seconds ago)
  has history:          yes
  has statistics:       yes
  battery
    present:             yes
    rechargeable:        yes
    state:               fully-charged
    warning-level:       none
    energy:              20.95 Wh
    energy-empty:        0 Wh
    energy-full:         20.95 Wh
    energy-full-design:  24 Wh
    energy-rate:         0 W
    voltage:             12.489 V
    percentage:          100%
    capacity:            87.2917%
    technology:          lithium-polymer
    icon-name:          'battery-full-charged-symbolic'
```
```
gabriel@gryphon:~> env LANG=C upower -d
Device: /org/freedesktop/UPower/devices/line_power_AC
  native-path:          AC
  power supply:         yes
  updated:              Wed Aug  8 02:24:52 2018 (35759 seconds ago)
  has history:          no
  has statistics:       no
  line-power
    warning-level:       none
    online:              yes
    icon-name:          'ac-adapter-symbolic'

Device: /org/freedesktop/UPower/devices/battery_BAT0
  native-path:          BAT0
  vendor:               SMP
  model:                01AV421
  serial:               1162
  power supply:         yes
  updated:              Wed Aug  8 12:18:52 2018 (119 seconds ago)
  has history:          yes
  has statistics:       yes
  battery
    present:             yes
    rechargeable:        yes
    state:               fully-charged
    warning-level:       none
    energy:              20.95 Wh
    energy-empty:        0 Wh
    energy-full:         20.95 Wh
    energy-full-design:  24 Wh
    energy-rate:         0 W
    voltage:             12.489 V
    percentage:          100%
    capacity:            87.2917%
    technology:          lithium-polymer
    icon-name:          'battery-full-charged-symbolic'

Device: /org/freedesktop/UPower/devices/battery_BAT1
  native-path:          BAT1
  vendor:               SANYO
  model:                01AV427
  serial:               3199
  power supply:         yes
  updated:              Wed Aug  8 12:20:06 2018 (45 seconds ago)
  has history:          yes
  has statistics:       yes
  battery
    present:             yes
    rechargeable:        yes
    state:               fully-charged
    warning-level:       none
    energy:              69.12 Wh
    energy-empty:        0 Wh
    energy-full:         69.2 Wh
    energy-full-design:  71.28 Wh
    energy-rate:         2.4 W
    voltage:             12.448 V
    percentage:          100%
    capacity:            97.0819%
    technology:          lithium-ion
    icon-name:          'battery-full-charged-symbolic'

Device: /org/freedesktop/UPower/devices/DisplayDevice
  power supply:         yes
  updated:              Wed Aug  8 04:02:06 2018 (29925 seconds ago)
  has history:          no
  has statistics:       no
  battery
    present:             yes
    state:               fully-charged
    warning-level:       none
    energy:              90.07 Wh
    energy-full:         90.15 Wh
    energy-rate:         2.4 W
    percentage:          99.9113%
    icon-name:          'battery-full-charged-symbolic'

Daemon:
  daemon-version:  0.99.7
  on-battery:      no
  lid-is-closed:   no
  lid-is-present:  yes
  critical-action: HybridSleep
```

Will test/look into it later if that's not enough or if you don't have the time.
